platform/linux-generic/pktio/loop.c

```diff
@@ -173,6 +176,41 @@ static int loopback_recv(pktio_entry_t *pktio_entry, int index ODP_UNUSED,
 	return num_rx;
 }
 
+#define OL_TX_CHKSUM_PKT(_cfg, _capa, _proto, _ovr_set, _ovr) \
+	(_capa && _proto && (_ovr_set ? _ovr : _cfg))
+
+static inline int check_proto(void *l3_hdr,
+			      uint32_t l3_len,
+			      odp_bool_t *l3_proto_v4,
+			      uint8_t *l4_proto)
+{
+	uint8_t l3_proto_ver = _ODP_IPV4HDR_VER(*(uint8_t *)l3_hdr);
+
+	if (l3_proto_ver == _ODP_IPV4 && l3_len >= _ODP_IPV4HDR_LEN) {
+		_odp_ipv4hdr_t *ip = l3_hdr;
+
+		if (_ODP_IPV4HDR_IS_FRAGMENT(_odp_be_to_cpu_16(ip->frag_offset)))
+			return -1;
```

Comment:
If it is an IPv4 fragment, we should still be able to compute IPv4 csum (but not UDP/TCP csum).
Use (as for dpdk pktio):
*l3_proto_v4 =1;
*l4_proto = 0;

Reply:
ack, thanks